myst_parser/docutils_.py

```diff
@@ -1,36 +1,157 @@
+) -> Tuple[dict, str]:
+    """Convert an ``MdParserConfig`` attribute into a Docutils optparse options dict."""
+    if at.type is int:
+        return {"validator": _validate_int}, f"(type: int, default: {default})"
+    if at.type is bool:
+        return {
+            "validator": frontend.validate_boolean
+        }, f"(type: bool, default: {default})"
+    if at.type is str:
+        return {}, f"(type: str, default: '{default}')"
+    if at.type == Iterable[str] or at.name == "url_schemes":
+        return {
+            "validator": frontend.validate_comma_separated_list
+        }, f"(type: comma-delimited, default: '{','.join(default)}')"
+    if at.type == Tuple[str, str]:
+        return {
+            "validator": _create_validate_tuple(2)
+        }, f"(type: str,str, default: '{','.join(default)}')"
+    if at.type == Union[int, type(None)] and at.default is None:
+        return {
+            "validator": _validate_int,
+            "default": None,
+        }, f"(type: null|int, default: {default})"
+    if at.type == Union[Iterable[str], type(None)] and at.default is None:
+        return {
+            "validator": frontend.validate_comma_separated_list,
+            "default": None,
+        }, f"(type: comma-delimited, default: '{default or ','.join(default)}')"
+    raise AssertionError(
+        f"Configuration option {at.name} not set up for use in docutils.conf."
+        f"Either add {at.name} to docutils_.DOCUTILS_EXCLUDED_ARGS,"
+        "or add a new entry in _docutils_optparse_of_attribute."
+    )
 
-    - A sequence of option tuples
-    """
+
+def _docutils_setting_tuple_of_attribute(
+    attribute: Attribute, default: Any
+) -> Tuple[str, Any, Any]:
+    """Convert an ``MdParserConfig`` attribute into a Docutils setting tuple."""
+    name = f"myst_{attribute.name}"
+    flag = "--" + name.replace("_", "-")
+    options = {"dest": name, "default": DOCUTILS_UNSET}
+    at_options, type_str = _docutils_optparse_options_of_attribute(attribute, default)
+    options.update(at_options)
+    help_str = attribute.metadata.get("help", "") if attribute.metadata else ""
+    return (f"{help_str} {type_str}", [flag], options)
+
+
+def _myst_docutils_setting_tuples():
+    """Return a list of Docutils setting for the MyST section."""
+    defaults = MdParserConfig()
+    return tuple(
+        _docutils_setting_tuple_of_attribute(at, getattr(defaults, at.name))
+        for at in MdParserConfig.get_fields()
+        if at.name not in DOCUTILS_EXCLUDED_ARGS
+    )
+
+
+def create_myst_config(settings: frontend.Values):
+    """Create a ``MdParserConfig`` from the given settings."""
+    values = {}
+    for attribute in MdParserConfig.get_fields():
+        if attribute.name in DOCUTILS_EXCLUDED_ARGS:
+            continue
+        setting = f"myst_{attribute.name}"
+        val = getattr(settings, setting, DOCUTILS_UNSET)
+        with suppress(AttributeError):
+            delattr(settings, setting)
```

Comment:
@cpitclaudel why do you need to delete the setting here?

Reply:
(note here I made the getting/deleting optional, since if you include a MyST file from an RST file, the MyST settings will not have been loaded)

Reply:
I forget the details :/ I *think* it was to avoid namespace pollution? I'm guessing I was worried about having the settings both directly in the document under `myst_*` and in a config object, so I was planning to remove all the `myst_*` settings from `document.settings` and instead have a single object `document.settings.myst_config`.

Of course, this explanation would make more sense if I had then actually stored the config object as `document.settings.myst_config`, but the code doesn't seem to do that, so either I forgot or my guess above is wrong ^^

Reply:
I think I'll remove it for now then, as it doesn't appear to break anything